### DISCUSSION TITLE
PNP Search Results Opens Sharepoint List Item instead of URL associated with List Item

### DISCUSSION THREAD

**Discussion by dbcjlmsw** [microsoft-search/pnp-modern-search#4078] [Q&A]
**The Goal:**

1. Staff uses search box to find a SharePoint page associated for their department, team or program.
2. Search Results shows the Title of the Sharepoint Page.
3. Staff Clicks on the Title in the Search Result Box and it takes them directly to the Sharepoint page or webpage.

**The Background:**

1. Search Box is connected to the Search Results
2. Search Results is utilizing a Sharepoint List (Facility Directory) that has multiple columns. The two main columns that are of concern is the Title column and the URL column (Site URL) that shows the links for the item.

**Where I am So Far**

1. The Search Results from the List are showing up when searched in the Search Box.
2. The Current Query Text is {searchTerms} (Path:{Site} AND (ListId:List Id Numbers))
3. Result Source Id/ Scope| Name is LocalSharePointResults
4. Selected Properties are Facility Directory and Title
5. Trim Duplicates Selected
6. Under Layouts I have chosen Detailed List.
7. Under Available Connections Use input query text is toggled on.  I have chose Dynamic Value and the source is connected to the search box and properties is the search query.

**The Issue**
When I click on a search result title. It takes me to the sharepoint list displaying the list item. I want to click on it and it take me to the URL that is associated with the item in the results. I dont want it to take me to the Sharepoint list. 

What I have tried a few thing so far but I can not get it to open the associated URL instead of the Sharepoint List Item.

Any help would be greatly appreciated. Thank you for your time and assistance.

**Comment by kasperbolarsen** [comment]
I think you will get what you are looking for by changing the slot "Path" from the default value to Path
![image](https://github.com/user-attachments/assets/977133eb-69a9-4f72-963c-7aa634f8f828)

**Reply by dbcjlmsw** [reply]
Once I changed to Path the search results no longer became clickable.

**Reply by kasperbolarsen** [reply]
Please check that Path is in Selected Fields

**Reply by dbcjlmsw** [reply]
If you mean selected properties. It is in there as well.

**Reply by kasperbolarsen** [reply]
But it sounds like it doesn't contain any value. Can you verify , using either the debug layout or one of the tools here, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Reply by dbcjlmsw** [reply]
It is showing null.

**Reply by kasperbolarsen** [reply]
Any of the other managed properties that you can use since Path for some weird reason is null? (it really shouldn't)

**Reply by dbcjlmsw** [reply]
Oh let me rephrase that Path isn't null. 
Path is showing the URL for the display form of the list item.
That field that was showing null was the field holding the website url.

I dont want it to go to the sharepoint list.

I want it to go to the Sharepoint URL that is one of fields of the list. 
Maybe I am not explaining it properly.

For example. I have a Sharepoint List called "Hospitals in my State". I have a column in the sharepoint list called Hospital Websites.
The list has been added to my Search Box and Search Results
I want to search for ABC Hospital.
I put ABC Hospital in the search box.
The search result for ABC Hospital comes up.

**What is happening**
When I click ABC Hospital, it take take me to the sharepoint display form for ABC Hospital.

**_I dont want that._** 

**What I want to happen**
When I click ABC hospital I want it to redirect me to the URL in the column Hospital Websites associated with ABC Hospital.

However I dont want it to show the full URL/ path. I want it to show the Hosptial Name but take me to the Hospital Website.


**_I hope that clarifies things. I apologize if I'm not making sense. I am learning this as I go. My knowledge has been Youtube videos and Share Point Open Hours calls. Thank you again for your time and assistance. I really appreciate it._**.

**Reply by kasperbolarsen** [reply]
I think I understand your requirement now :-)
When you entered a value in the field Hospital Websites the search crawler automatically created two crawled properties, something like ows_Hospital%20Websites and ows_q_text_Hospital%20Websites.
If you had created the Hospital Websites field as a SITE column, the search crawler would also have created a Managed Property with a name like Hospital%20WebsitesOWSTEXT
If you created the field as a LIST column, the Managed Property is NOT created

Once you have a Managed Property (we can't use Crawled Properties for anything but mapping them to Managed Properties) you can add that Managed Property to Selected Properties in the Results web part, add it in Slots and then, finally you can use Manage Columns to update the Handlebars expression, by changing {{slot item @root.slots.Path}} to the {{slot item @root.slots.NAMEOFTHESLOTYOUJUSTADDED}}

![image](https://github.com/user-attachments/assets/f4d5a518-c26c-448c-ac75-00b97a6fcc9a)

**Reply by dbcjlmsw** [reply]
Next question. Do I map it using a refinable string?
